Assets/Scripts/Controllers/CharacterController.cs

```diff
@@ -23,6 +32,28 @@ private void FixedUpdate()
     }
 
 
+    private void Update()
+    {
+        // Check if the player has fired a weapon recently
+        if (_fireRate == 0)
+        {
+            if(Input.GetButton("Fire1"))
+            {
+                //Get values necessary to spawn projectiles outside of player collision
+                _projectileSpawnModifier = Input.mousePosition - _cam.WorldToScreenPoint(_rb.position);
+                _projectileSpawnModifier.Normalize();
+                _projectileSpawnModifier = _projectileSpawnModifier * _playerBoundingBox.size;
```

Comment:
There are some good methods for preventing self-collision with the player's own projectiles, but this is pretty clever.

Also consider using [Collision Layers](https://docs.unity3d.com/Manual/LayerBasedCollision.html) to prevent self-collision.

No action needed.